### PR TITLE
Backport period rounding

### PR DESCRIPTION
**Pull request by rofinn**

### CODE REVIEW

**Comment by martinholters** [comment]
Should add an entry to the README.

**Comment by rofinn** [comment]
Is this good to merge?

File: README.md [hunk 1]
```diff
@@ -235,6 +235,9 @@ Currently, the `@compat` macro supports the following syntaxes:
   an array (respectively), and indexing such objects allows translating from one kind of index
   to the other ([#25113]).
 
+* `Dates.Period` rounding (e.g., `round(Dates.Hour(36), Dates.Day, RoundNearestTiesUp) == Dates.Day(2)` ([##24182]).
+
+
```
Comment (omus) [inline README.md]: Don't you need to add the link at the bottom of this file? Also should reference should only have one `#`

**Comment by rofinn** [comment]
Looks like all of the builds are erroring on linux nightly with `MethodError: no method matching Base.HasShape()`.

**Pull request merged**